packages/sanes-browser-extension/src/index.tsx

```diff
@@ -34,20 +34,18 @@ const render = (
   );
 };
 
-type PersonaContexType = [GetPersonaResponse, boolean];
+// REVIEW getHasStoredPersona() has a Promise<boolean> signature but
+// hasStoredPersona gets type boolean | null
+Promise.all([getPersonaData(), getHasStoredPersona()]).then(([persona, hasStoredPersona]) => {
```

Comment:
It all gets much simpler by avoiding `Promise.all`. There is no need to run those in parallel:

```ts
(async () => {
  const persona = await getPersonaData();
  const hasStoredPersona = await getHasStoredPersona();

  // use unchanged old code
})();
```

Reply:
Why not to run it in parallel?

Reply:
Aha! There is a much simpler fix: Replace 

```
Promise.all([getPersonaData(), getHasStoredPersona()]).then(
```

with

```
Promise.all([getPersonaData(), getHasStoredPersona()] as const).then(
```

The `as const` ensure the array literal is interpreted as a 2-component array. For fixed length arrays, the types can be different in each component.